var/spack/repos/builtin/packages/siesta/package.py

```diff
@@ -63,22 +64,33 @@ def configure(self, spec, prefix):
                           # Intel's mpiifort is not found
                           'MPIFC=%s' % spec['mpi'].mpifc
                           ]
-        with working_dir('Obj'):
-            sh('../Src/configure', *configure_args)
-            sh('../Src/obj_setup.sh')
-        with working_dir('Obj_trans', create=True):
-            sh('../Src/configure', *configure_args)
-            sh('../Src/obj_setup.sh')
+        for dir, create in [('Obj', False), ('Obj_trans', True)]:
```

Comment:
To make things simpler, always `create=True`. If the directory already exists, it won't do anything.